### ISSUE TITLE
Label ignores visible_when

### ISSUE THREAD

**Issue by jonathanrocher** [enthought/traitsui#298]
In the following snippet, I would expect the label not to show, but it shows:

```
from traits.api import HasTraits, Int, Str
from traitsui.api import View, Item, Label, HGroup

class A(HasTraits):
    x = Int

    view = View(
        Label("This is a message in a group that shouldn't show.",
               visible_when="x != 0"),
        Item('x'),
        resizable=True
    )

A(x=0).configure_traits()
```

As a workaround, embedding the Label in a group makes it work. (And it also looks nicer because there is some padding around the message.)

**Comment by corranwebster** [comment]
Not a known bug, and _should_ in principle work (`Label` is just a subclass of `Item`), so the fact that it doesn't is a bug.  Happens in both Wx and Qt toolkits.  Don't have time to look into it more right now.

Thanks for the report.

**Comment by itziakos** [comment]
`Label` also ignores `enabled_when` for similar reasons. 

*Note: I do not thing that fixing this is easy*

**Comment by tmreay** [comment]
Ran in to this issue in the TEL project. A fix would be nice, but looking how the `Label.__init__` works, I agree that it's not going to be easy. We will wrap in a `Group` as a work around.

**Comment by rahulporuri** [comment]
another example to reproduce this error 

```
from traits.api import *
from traitsui.api import *

class Test(HasTraits):

    show = Bool(True)

    button = Button("Hide")

    traits_view = View(
        Label("I refuse to hide!", visible_when="show"),
        UItem("button"),
    )

    def _button_fired(self):
        self.show = False


if __name__ == "__main__":
    Test().configure_traits()
```

even if we are unable to fix this issue immediately, we should create a commonly known issues page in the documentation and mention this there.

**Comment by rahulporuri** [comment]
A workaround that doesn't need the use of a `Group` wrapping the `Label` is. Thanks @pedrorivotti 

```
from traits.api import *
from traitsui.api import *

class Test(HasTraits):
    show = Bool(True)
    button = Button("Hide")
    _dummy_text = Str("I refuse to hide!")

    traits_view = View(
        UItem("_dummy_text", style="readonly", visible_when="show"),
        UItem("button"),
    )

    def _button_fired(self):
        self.show = False

if __name__ == "__main__":
    Test().configure_traits()
```

**Comment by aaronayres35** [comment]
closing as this was fixed in #1544
